src/storage.ts

```diff
@@ -123,10 +127,44 @@ export class LlamaStorage {
   }
 
   async list(): Promise<FileRecord[]> {
-    return this.db.getAll('metadata');
+    const localMeta = this.db.getAll('metadata');
+    let firebaseList: firebase.ListResult | null = null;
+    if (firebaseStorage) {
+      firebaseList = await firebase.listAll(firebase.ref(firebaseStorage, firebaseDir));
```

Comment:
Is it the right thing to do to sync this often?

An alternative approach to data consistency, having a slightly weaker contract, might be to have the local state be leading, but do an occasional sync w/ Firebase to identify differences (e.g. once per launch in the background, or using Periodic Background Sync). That might be kinder to offline users as well, and those on lie-fi.

Reply:
> Is it the right thing to do to sync this often?

Probably not - I'm mostly just figuring out how to make Firebase work at all for now, and worrying about efficiency later. Hooray for tech debt.

> (e.g. once per launch in the background, or using Periodic Background Sync).

Never used Periodic Background Sync, but it does sound reasonable. https://github.com/GoogleChromeLabs/llaminator/issues/110